database/skylink.go

```diff
@@ -0,0 +1,99 @@
+package database
+
+import (
+	"context"
+
+	accdb "github.com/SkynetLabs/skynet-accounts/database"
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var (
+	// ErrInvalidSkylink is returned when a client call supplies an invalid
+	// skylink hash.
+	ErrInvalidSkylink = errors.New("invalid skylink")
+	// ErrSkylinkExists is returned when we try to create a skylink that already
+	// exists.
+	ErrSkylinkExists = errors.New("skylink already exists")
+	// ErrSkylinkNoExist is returned when we try to get a skylink that doesn't
+	// exist.
+	ErrSkylinkNoExist = errors.New("skylink does not exist")
+)
+
+type (
+	// Skylink represents a skylink object in the DB.
+	Skylink struct {
+		ID      primitive.ObjectID `bson:"_id,omitempty"`
+		Skylink string             `bson:"skylink"`
+		Servers []string           `bson:"servers"`
+	}
+)
+
+// SkylinkCreate inserts a new skylink into the DB. Returns an error if it
+// already exists.
+func (db *DB) SkylinkCreate(ctx context.Context, sl string, server string) (Skylink, error) {
+	if !accdb.ValidSkylinkHash(sl) {
```

Comment:
kind of confused with the naming here, this will check whether it's a valid base32 or base64 encoded skylink string right? how is this different from the loadstring validation using the skylink type?

Reply:
replaced by `LoadString`.